packages/utils/src/EventHandlers/EventHandlers.ts

```diff
@@ -0,0 +1,147 @@
+import { ConnectorRegistry } from './ConnectorRegistry';
+import {
+  EventHandlerUpdates,
+  CraftEventListener,
+  EventHandlerConnectors,
+  CraftDOMEvent,
+} from './interfaces';
+import { isEventBlockedByDescendant } from './isEventBlockedByDescendant';
+
+export abstract class EventHandlers<O extends Record<string, any> = {}> {
+  options: O;
+
+  private registry: ConnectorRegistry = new ConnectorRegistry();
+  private subscribers: Set<(msg: EventHandlerUpdates) => void> = new Set();
+
+  constructor(options?: O) {
+    this.options = options;
+  }
+
+  listen(cb: (msg: EventHandlerUpdates) => void) {
+    this.subscribers.add(cb);
+    return () => this.subscribers.delete(cb);
+  }
+
+  disable() {
+    this.registry.disable();
+
+    this.subscribers.forEach((listener) => {
+      listener(EventHandlerUpdates.HandlerDisabled);
+    });
+  }
+
+  enable() {
+    this.registry.enable();
+
+    this.subscribers.forEach((listener) => {
+      listener(EventHandlerUpdates.HandlerEnabled);
+    });
+  }
+
+  cleanup() {
+    this.disable();
+    this.subscribers.clear();
+    this.registry.clear();
+  }
+
+  addCraftEventListener<K extends keyof HTMLElementEventMap>(
+    el: HTMLElement,
+    eventName: K,
+    listener: CraftEventListener<K>,
+    options?: boolean | AddEventListenerOptions
+  ) {
+    const bindedListener = (e: CraftDOMEvent<HTMLElementEventMap[K]>) => {
+      // Store initial Craft event value
+      if (!e.craft) {
+        e.craft = {
+          blockedEvents: {},
+          stopPropagation: () => {},
+        };
+      }
+
+      if (!isEventBlockedByDescendant(e, eventName, el)) {
```

Comment:
i wonder if this would be better named `wasHandledByDescendant` ? 🤔

Reply:
My rationale is that it's easier to reason about these events as `handled` or `consumed` rather than `blocked` since blocked makes me feel like they were prevented from occurring somewhat 😅

Reply:
> blocked makes me feel like they were prevented from occurring somewhat 

That's because they're being blocked. The descendant calls `.craft.stopPropagation` which blocks the event from further propagating to an ancestor.

Reply:
Yeah i get it, but it's the propagation that is stopped, not the event itself, thats why i found the name confusing

Reply:
I'm not sure if I follow. If the propagation is stopped, the parent event should by all means not fire - which is what we're trying to emulate here, inside the parent event handler.